osu.Game.Tests/OnlinePlay/MultiplayerPlaylistItemTest.cs

```diff
@@ -0,0 +1,66 @@
+        {
+            var faker = new Faker<MultiplayerPlaylistItem>()
+                        .StrictMode(true)
+                        .RuleFor(o => o.ID, f => f.Random.Long())
+                        .RuleFor(o => o.OwnerID, f => f.Random.Int())
+                        .RuleFor(o => o.BeatmapID, f => f.Random.Int())
+                        .RuleFor(o => o.BeatmapChecksum, f => f.Random.Hash())
+                        .RuleFor(o => o.RulesetID, f => f.Random.Int())
+                        .RuleFor(o => o.RequiredMods, f => f.Make(5, _ => new APIMod { Acronym = f.Random.String2(3) }))
+                        .RuleFor(o => o.AllowedMods, f => f.Make(5, _ => new APIMod { Acronym = f.Random.String2(3) }))
+                        .RuleFor(o => o.Expired, f => f.Random.Bool())
+                        .RuleFor(o => o.PlaylistOrder, f => f.Random.UShort())
+                        .RuleFor(o => o.PlayedAt, f => f.Date.RecentOffset())
+                        .RuleFor(o => o.StarRating, f => f.Random.Double())
+                        .RuleFor(o => o.Freestyle, f => f.Random.Bool());
+
+            for (int i = 0; i < 100; i++)
+            {
+                MultiplayerPlaylistItem item = faker.Generate();
+                Assert.That(MessagePackSerializer.SerializeToJson(item.Clone()), Is.EqualTo(MessagePackSerializer.SerializeToJson(item)));
+            }
+        }
+
+        [Test]
+        public void TestConstructFromAPIModel()
+        {
+            var faker = new Faker<MultiplayerPlaylistItem>()
+                        .StrictMode(true)
+                        .RuleFor(o => o.ID, f => f.Random.Long())
+                        .RuleFor(o => o.OwnerID, f => f.Random.Int())
+                        .RuleFor(o => o.BeatmapID, f => f.Random.Int())
+                        .RuleFor(o => o.BeatmapChecksum, f => f.Random.Hash())
+                        .RuleFor(o => o.RulesetID, f => f.Random.Int())
+                        .RuleFor(o => o.RequiredMods, f => f.Make(5, _ => new APIMod { Acronym = f.Random.String2(3) }))
+                        .RuleFor(o => o.AllowedMods, f => f.Make(5, _ => new APIMod { Acronym = f.Random.String2(3) }))
+                        .RuleFor(o => o.Expired, f => f.Random.Bool())
+                        .RuleFor(o => o.PlaylistOrder, f => f.Random.UShort())
+                        .RuleFor(o => o.PlayedAt, f => f.Date.RecentOffset())
+                        .RuleFor(o => o.StarRating, f => f.Random.Double())
+                        .RuleFor(o => o.Freestyle, f => f.Random.Bool());
+
+            for (int i = 0; i < 100; i++)
+            {
+                MultiplayerPlaylistItem initialItem = faker.Generate();
+                MultiplayerPlaylistItem copiedItem = new MultiplayerPlaylistItem(new PlaylistItem(initialItem));
+                Assert.That(MessagePackSerializer.SerializeToJson(copiedItem), Is.EqualTo(MessagePackSerializer.SerializeToJson(initialItem)));
```

Comment:
How did you find this `SerializeToJson()` method? Is this even meant for usage by external consumers and not some really-internal-but-public-right-now helper? (I tried blame on the messagepack repo and came up short.)

Which is to say, I'd prefer this did equality assertions on the binary messagepack content, probably.

Reply:
I found it by intellisense.

I originally did binary comparisons but it didn't lead to usable error messages when I actually had to resolve an error, and I thought to rather use something annotated by the models than straight Newtonsoft or System.Text.Json that may have unknown behaviours...

Is this significant enough to address for you?

Reply:
I guess it's probably fine until it's not... At least in the messagepack repo the method seems to have been living there for a good few years so maybe the risk of it disappearing is low.